code/src/components/app-range-slider/AppRangeSlider.tsx

```diff
@@ -1,114 +1,99 @@
-import { ChangeEvent, useState } from "react";
+import { ChangeEvent } from "react";
 
 import Slider from "@mui/material/Slider";
 
 import AppBox from "@/components/app-box/AppBox";
 import AppInput from "@/components/app-input/AppInput";
 import { AppRangeSliderProps } from "@/components/app-range-slider/AppRangeSlider.types";
+import parseRangeValues from "@/components/app-range-slider/utils/parse-range-values/parseRangeValues";
 import AppTypography from "@/components/app-typography/AppTypography";
 
 import cn from "@/utils/cn/cn";
 
 import "@/components/app-range-slider/AppRangeSlider.scss";
 
-const parseRangePositionValue = (
-  event: ChangeEvent<HTMLInputElement>,
-  limiterFn: typeof Math.max | typeof Math.min,
-  edgeValue: number
-) => {
-  const value = event.target.value;
-  const numericValue = parseInt(value);
-  const limitedNewValue = limiterFn(edgeValue, numericValue);
-
-  // needed for typescript, we don't need to pass first parameter at all
-  const typedEvent = event as unknown as Event;
-
-  return { typedEvent, limitedNewValue };
-};
-
 const AppRangeSlider = ({
   className,
   onChange,
   min = 0,
   max = 20000,
-  step = 10,
+  step,
   value,
   ...props
 }: AppRangeSliderProps) => {
-  const initialRangeStart = value?.[0] ?? min;
-  const initialRangeEnd = value?.[1] ?? max;
-
-  const [rangeStart, setRangeStart] = useState(initialRangeStart);
-  const [rangeEnd, setRangeEnd] = useState(initialRangeEnd);
+  const rangeStart = value?.[0].toString() ?? "";
+  const rangeEnd = value?.[1].toString() ?? "";
+
+  const { sliderRange, inputData } = parseRangeValues({
+    rangeStart,
+    rangeEnd,
+    min,
+    max
+  });
+
+  const updateWithValue = <Value extends string | number>(value: Value[]) => {
```

Comment:
I think it will be better
```suggestion
  const updateWithValue = (value: (string | number)[]) => {}
```

Reply:
Thanks Dmytro, it was very useful. I used your suggestion